poller.py

```diff
@@ -73,50 +59,71 @@ def _process_input(self, input_q: dict) -> None:
+                                                                 layer_id,
+                                                                 chat_ids)) for layer_id, chats_batch in enumerate(batched_chats))
+        await asyncio.gather(*tasks)
+
+    async def _process_chats_batch(self, chats_batch: List[str], layer_id: int, chat_ids: List[int]) -> None:
+        utts_batch = [(chat_ids[utt_id], utt) for utt_id, utt in enumerate(chats_batch) if utt]
+        j = self.config['infer_batch_length']
+        chunked_utts_batch = [utts_batch[i * j:(i + 1) * j] for i in range((len(utts_batch) + j - 1) // j)]
+        await asyncio.gather(*(self.loop.create_task(self._process_chunk(chunk, layer_id)) for chunk in chunked_utts_batch))
+
+    async def _process_chunk(self, chunk: List[Tuple[int, str]], layer_id: int) -> None:
+        ids_utts_batch = list(zip(*chunk))
+        ids_batch = list(ids_utts_batch[0])
+        utts_batch = list(ids_utts_batch[1])
+        data = {self.config["model_param_name"]: utts_batch}
+        try:
+            response = await self.loop.run_in_executor(None, functools.partial(requests.post,
+                                                                               self.config['model_url'],
+                                                                               json=data,
+                                                                               timeout=self.config['request_timeout']))
+        except requests.exceptions.ReadTimeout:
+            response = requests.Response()
+            response.status_code = 503
+
+        if response.status_code == 200:
+            zip_batch = zip(ids_batch, response.json())
+        else:
+            self.log.error(f'Got {response.status_code} code from {self.config["model_url"]}')
+            zip_batch = zip_longest(ids_batch, [], fillvalue='Server error')
+        tasks = (self.loop.create_task(self._send_results(chat_id, chat_resp, layer_id)) for (chat_id, chat_resp) in zip_batch)
 
-                for resp in zip(ids_batch, response_batch):
-                    self._send_results(*resp)
+        await asyncio.gather(*tasks)
 
     @staticmethod
-    def _process_message_text(message_text: str) -> Optional[str]:
-        if message_text[:6] == '/start' or message_text[:4] == '/end':
-            processed_message = None
+    def _process_payload(payload: Dict) -> Optional[str]:
+        if payload.get('command', None) is not None:
+            message = None
         else:
-            processed_message = message_text
-
-        return processed_message
-
-    def _send_results(self, chat_id, response) -> None:
-        resp_text = str("{\"text\":\"" + str(response) + "\"}")
+            message = payload['text']
+        return message
 
+    async def _send_results(self, chat_id: int, response: list, layer_id: int) -> None:
+        buf = {'text': ' '.join(str(element) for element in response)}
```

Comment:
If model's response is `['']` (for example ODQA could send such response) then resp_text (see below) equals `{'text': ''}`. Router bot not sending any message to user with empty `text` field. What better to send in such cases? Maybe `-` to give user information that model respond nothing and not just ignored last message?

Reply:
Not need for now, I think, we should simply translate model answers until router-bot user say to change this behavior.